libraries/AP_Proximity/AP_Proximity.cpp

```diff
@@ -208,36 +91,117 @@ AP_Proximity::AP_Proximity()
 
 // initialise the Proximity class. We do detection of attached sensors here
 // we don't allow for hot-plugging of sensors (i.e. reboot required)
-void AP_Proximity::init(void)
+void AP_Proximity::init()
 {
     if (num_instances != 0) {
         // init called a 2nd time?
         return;
     }
-    for (uint8_t i=0; i<PROXIMITY_MAX_INSTANCES; i++) {
-        detect_instance(i);
-        if (drivers[i] != nullptr) {
+
+    // instantiate backends
+    uint8_t serial_instance = 0;
+    for (uint8_t instance=0; instance<PROXIMITY_MAX_INSTANCES; instance++) {
+        switch (get_type(instance)) {
+        case Type::None:
+            return;
```

Comment:
This should probably be "break" otherwise we can't load PRX2_ without having configured PRX1

Reply:
txs, nice catch. fixed